### PR TITLE
add ability to use nbt in icon

### PR DESCRIPTION
**Pull request by Minecraftschurli**
This PR adds the ability to use nbt with the command syntax in the `itemIcon`/`catalogueImageIcon` field

### CODE REVIEW

File: src/main/java/com/mrcrayfish/catalogue/client/screen/CatalogueModListScreen.java [hunk 1]
```diff
@@ -755,31 +759,27 @@ private Item getItemIcon()
...
-                        ITEM_CACHE.put(this.info.getModId(), item);
-                        return item;
-                    }
-                }
+                try {
+                    ItemParser parser = new ItemParser(new StringReader(itemIcon), false).parse();
+                    ItemStack item = new ItemStack(parser.getItem(), 1, parser.getNbt());
+                    ITEM_CACHE.put(this.info.getModId(), item);
+                    return item;
+                } catch (CommandSyntaxException ignored) {}
```
Comment (MrCrayfish) [inline src/main/java/com/mrcrayfish/catalogue/client/screen/CatalogueModListScreen.java]: Just fix this code style and then it'll be ready to merge.

Comment (Minecraftschurli) [reply src/main/java/com/mrcrayfish/catalogue/client/screen/CatalogueModListScreen.java]: Should I do 
```java
catch (CommandSyntaxException ignored)
{
}
```
or
```java
catch (CommandSyntaxException ignored) {}
```
for the style fix?

**Comment by MrCrayfish** [comment]
Going to withhold merging this until 1.18 since it's a breaking change (e.g. loading a nbt version on an older version either 1.16 or 1.16). Once it's 1.18 it won't matter since every version of will have nbt support.

**Comment by Minecraftschurli** [comment]
I wouldn't call it breaking since the new format just extends the old (just the resource location is still valid) and the changes are only internal.

**Comment by MrCrayfish** [comment]
It is breaking because old versions of the mod won't be able to understand the NBT format. I understand that this PR can read the old definitons like normal, let's just say someone is using an older version of the Catalogue (that doesn't support nbt) and they install a mod that uses the nbt support.

**Comment by Minecraftschurli** [comment]
Then it would default to the way it is if no itemIcon is specified

**Comment by Minecraftschurli** [comment]
Retargeting to 1.18

**Pull request closed without merging**